File: cms/src/main/java/com/insighthub/cms/service/PostService.java
package com.insighthub.cms.service;
import com.insighthub.cms.dto.PostRequest;
import com.insighthub.cms.dto.PostResponse;
import java.util.List;
public interface PostService {
    PostResponse createPost(PostRequest request,String authorEmail);
    List<PostResponse> getAllPosts();
    PostResponse getPostBySlug(String slug);
    PostResponse updatePost(Long postId,PostRequest request,String userEmail);
    void deletePost(Long postId,String userEmail);
}

File: insight-hub-frontend/src/services/postService.js
import api from "../api/axios";

export async function getAllPosts() {
  const response = await api.get("/posts");
  return response.data;
}

export async function getPostBySlug(slug) {
  const response = await api.get(`/posts/${slug}`);
  return response.data;
}

export async function createPost(postData) {
  const response = await api.post("/posts", postData);
  return response.data;
}

export async function updatePost(postId, postData) {
  const response = await api.put(`/posts/${postId}`, postData);
  return response.data;
}

export async function deletePost(postId) {
  await api.delete(`/posts/${postId}`);
}
export async function submitForModeration(postId) {
  const response = await api.post(`/posts/${postId}/submit`);
  return response.data;
}